### ISSUE TITLE
Explore padding for improved JIT

### ISSUE THREAD

**Issue by quattro** [mancusolab/sushie#22]
Our current code uses JIT, but is suboptimal due to inconsistent shapes across groups (generally). One workaround could be to pad the smallest groups with 0s to match the largest sample size. This would enable the use of ndarrays throughout, without the need to use loops/lists at the cost of increased memory. In this setting JIT should be much more efficient, but it's unclear how much more efficient compared with our current implementation (or until they allow dynamic shapes for JIT).

- [x] Padding
- [ ] JAXSParse support

**Comment by quattro** [comment]
I think the combination of padding + `jax.experimental.sparse.BCOO` could really pay off in terms of speed if we provide a careful implementation.

**Comment by zeyunlu** [comment]
Done!
